packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -197,14 +197,6 @@ function ackSegment(
 				type: op.type === MergeTreeDeltaType.REMOVE ? "setRemove" : "sliceRemove",
 			};
 			segment.removes[segment.removes.length - 1] = removeStamp;
-
```

Comment:
we only need to do this bit once per obliterate, not per segment. I added a similar class of stuff to ObliterateInfo, and moving this logic to `MergeTree.ackOp` (renamed from `ackPendingSegment` as it's a bit clearer) allows some more strict validation. It also makes the "ack sliceRemove segment" operation work exactly the same as set removal now, and it's just some extra things we need to do at the per-op-ack level :)